webapp/src/sui.tsx

```diff
@@ -90,55 +91,87 @@ export class UiElement<T extends WithPopupProps> extends data.Component<T, {}> {
+        this.child("").dropdown("hide")
+    }
+
     componentDidMount() {
         this.popup()
         this.child("").dropdown({
-            action: (text: string, value: any, element: any) => {
-                let htmlElement: HTMLElement
+            action: (text: string, value: any, element: JQuery) => {
+                this.close()
 
-                this.child("").dropdown("hide")
-
-                // When we use the keyboard, it is not an HTMLElement that we receive, but a JQuery. Activating click on it reproduce the same behavior than a real click.
+                // When we use the keyboard, it is not an HTMLElement that we receive, but a JQuery.
                 if (typeof element.get === "function") {
-                    let jqueryElement = element as JQuery
-                    htmlElement = jqueryElement.get(0)
-                    htmlElement.click()
-                    return
-                } else {
-                    htmlElement = element as HTMLElement
-                }
-
-                if (htmlElement.tagName.toLowerCase() === 'a') {
-                    window.open((htmlElement as HTMLLinkElement).href, '_blank')
+                    if (element.get(0).tagName.toLowerCase() === 'a') {
+                        window.open((element.get(0) as HTMLLinkElement).href, '_blank')
+                    }
                 }
             },
             fullTextSearch: true,
-            onChange: (v: string) => {
+            onChange: (v: string, text: string, item: JQuery) => {
+                this.preventHide = true
+                item.get(0).focus()
+
                 if (this.props.onChange && v != this.props.value) {
                     this.props.onChange(v)
                 }
+            },
+            onShow: () => {
+                this.isOpened = true
+                this.forceUpdate()
+
+                var menuItems = this.child(".link")
+                menuItems.each((index: number, elem: HTMLElement) => {
+                    elem.onkeydown = this.handleTab
+                })
+            },
+            onHide: () => {
+                if (this.preventHide) {
+                    this.preventHide = false
+                    return false
+                }
+
+                this.isOpened = false
+                this.forceUpdate()
+                return true;
             }
         });
     }
 
     componentDidUpdate() {
         this.child("").dropdown("refresh")
         this.popup()
-        let hrefTags = ReactDOM.findDOMNode(this).getElementsByTagName("a")
-        for (let i = 0; i < hrefTags.length; i++) {
-            hrefTags.item(0).onclick = () => { return false }
-            hrefTags.item(0).onkeypress = () => { return false }
-        }
     }
 
     renderCore() {
         return (
             <div className={genericClassName("ui dropdown item", this.props) }
-                role={this.props.role}
+                role="menuitem"
                 title={this.props.title}
```

Comment:
Hum, we are in a DropdownMenuItem, except if we have a special case, I don't think we will need something else. At least for the moment I don't think we need.